src/raft/raft.go

```diff
@@ -595,7 +767,11 @@ func checkApplyEntries(rf *Raft) {
 		if rf.role == Leader {
 			newCommittedIndex := rf.committedIndex
 			for i := range rf.peers {
-				if rf.matchIndex[i] > newCommittedIndex {
+				// Leader can only commit logs whose term is the current term.
+				if rf.getPartialLogIndex(rf.matchIndex[i]) >= len(rf.logs.Entries) {
+					rf.debug("On Raft [%d], the real index [%d] for match index [%d] is out of bound ([%d])", i, rf.getPartialLogIndex(rf.matchIndex[i]), rf.matchIndex[i], len(rf.logs.Entries))
+				}
```

Comment:
Same here. This log seems to be added to solve a specific bug. Should remove it and only leave the general-purpose logs.